lib/fs.js

```diff
@@ -12,63 +12,68 @@ const DevMiddlewareError = require('./DevMiddlewareError');
 module.exports = {
   toDisk(context) {
     const compilers = context.compiler.compilers || [context.compiler];
+
     for (const compiler of compilers) {
-      compiler.hooks.afterEmit.tap('WebpackDevMiddleware', (compilation) => {
-        const { assets } = compilation;
-        const { log } = context;
-        const { writeToDisk: filter } = context.options;
-        let { outputPath } = compiler;
+      compiler.hooks.emit.tap('WebpackDevMiddleware', (compilation) => {
+        compiler.hooks.assetEmitted.tapAsync(
+          'WebpackDevMiddleware',
+          (file, content, callback) => {
+            let targetFile = file;
 
-        if (outputPath === '/') {
-          outputPath = compiler.context;
-        }
+            const queryStringIdx = targetFile.indexOf('?');
 
-        for (const assetPath of Object.keys(assets)) {
-          let targetFile = assetPath;
+            if (queryStringIdx >= 0) {
+              targetFile = targetFile.substr(0, queryStringIdx);
+            }
 
-          const queryStringIdx = targetFile.indexOf('?');
+            let { outputPath } = compiler;
 
-          if (queryStringIdx >= 0) {
-            targetFile = targetFile.substr(0, queryStringIdx);
-          }
+            // TODO Why? Need remove in future major release
```

Comment:
Can we remove this when we release the next version?

Reply:
Yes, i think we should drop this in next major, i keep this because i can't understand why it is here :confused: Any ideas?

Reply:
Yes, it's a good idea which is keeping. I don't know why... 
We'll drop this at other pr.